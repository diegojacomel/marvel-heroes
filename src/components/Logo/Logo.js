// Modules
import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.h1`
    margin: ${props => props.theme.spacing.none};
    line-height: 0.8;
`

const LogoWrapper = ({ children, title }) => (
    <Wrapper title={title}>
        {children}
    </Wrapper>
);

const Img = styled.img`
    height: ${props => props.theme.spacing.gt};
`

const ImgLogo = ({ src, alt }) => (
    <Img src={src} alt={alt} />
)

function Logo({ src, alt, title }) {
    return (
        <LogoWrapper title={title} id="testLogo">
            <ImgLogo src={src} alt={alt} id="testImgLogo" />
        </LogoWrapper>
    )
}

Logo.propTypes = {
    src: PropTypes.string,
    alt: PropTypes.string,
    title: PropTypes.string,
};

Logo.defaultProps = {
    src: '#',
    alt: 'Logo',
    title: 'Logo',
};

export default Logo;