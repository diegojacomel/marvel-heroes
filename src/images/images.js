// Images
import reactLogo from './react.png';
import marvel from './marvel.svg';

export default {
    reactLogo,
    marvel
}