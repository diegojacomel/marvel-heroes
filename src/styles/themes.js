// Themes
const primaryTheme = {
    colors: {
        primary: '#003366',
        secondary: '#000000',
        warning: '#000000',
        danger: '#d30d30',
        info: '#006633',
        default: '#dddddd',
        grayLight: '#f2f2f2',
        grayMedium: '#dddddd',
        grayDark: '#333333',
        black: '#000000',
        white: '#ffffff',
        hover: {
            primary: '#115394',
            secondary: '#000000',
            warning: '#000000',
            danger: '#d30d30',
            info: '#006633',
            default: '#dddddd',
            grayLight: '#f2f2f2',
            grayMedium: '#dddddd',
            grayDark: '#333333',
            black: '#000000',
            white: '#ffffff'
        }
    },
    spacing: {
        none: '0',
        xxs: '3px',
        xs: '5px',
        ss: '8px',
        sm: '10px',
        ssm: '15px',
        md: '20px',
        mmd: '25px',
        lg: '30px',
        llg: '35px',
        xg: '40px',
        xxg: '50px',
        gt: '75px',
        ggt: '100px',
        quarter: '25%',
        middle: '50%',
        full: '100%'
    }
}

export {
    primaryTheme
}