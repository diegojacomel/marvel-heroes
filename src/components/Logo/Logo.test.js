// Modules
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// Theme
import { primaryTheme } from '../../styles/themes'; 

// Images
import images from '../../images/images';

// Component
import Logo from './Logo';

describe('Logo', () => {
    const wrapper = shallow(<Logo />);

    test('It exist', () => {
        expect(wrapper.find('#testLogo')).toExist();
        expect(wrapper.find('#testImgLogo')).toExist();
    })

    test('At least one', () => {
        expect(wrapper).toContainMatchingElement('#testLogo');
        expect(wrapper).toContainMatchingElement('#testImgLogo');
    })

    test('Have display name', () => {
        expect(wrapper.find('#testLogo')).toHaveDisplayName('LogoWrapper');
        expect(wrapper.find('#testImgLogo')).toHaveDisplayName('ImgLogo');
    })

    test('Match selector', () => {
        expect(wrapper.find('#testLogo')).toMatchSelector('LogoWrapper');
        expect(wrapper.find('#testImgLogo')).toMatchSelector('ImgLogo');
    })

    test('Have props', () => {
        expect(wrapper.find('#testLogo')).toHaveProp('title');
        expect(wrapper.find('#testImgLogo')).toHaveProp('src');
        expect(wrapper.find('#testImgLogo')).toHaveProp('alt');
    })

    it('Render correctly', () => {
        const tree = renderer.create(
            <ThemeProvider theme={primaryTheme}>
                <Logo src={images.reactLogo} alt="alt" title="title" />
            </ThemeProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})