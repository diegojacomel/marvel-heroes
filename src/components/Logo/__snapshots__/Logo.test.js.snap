// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Render correctly 1`] = `
<h1
  className="sc-bdVaJa hxOzgB"
  title="title"
>
  <img
    alt="alt"
    className="sc-bwzfXH cinfRt"
    src="react.png"
  />
</h1>
`;
